### PR TITLE
rosette: Simplify some pointer macros.

### PR DESCRIPTION
**Pull request by nashef**
The only functional change in the PR is that the 'runtime-error' primitive now aborts the entire process, which causes it to exit to the shell with an actual error code. The rest of the changes are cleanups and non-operative changes to various macros.

### CODE REVIEW

**Comment by nashef** [comment]
This PR seems to have picked up some indent changes, probably due to someone (possibly me) not running `clang-format` before creating their pull request. My apologies.

File: rosette/src/Ob.cc [hunk 1]
```diff
@@ -76,16 +76,30 @@ int ESCTAG(pOb x) {
     return GET_ESCTAGGED_TAG(te);
 }
 
-int TAGVAL(pOb x) {
+/**
+ * NB(leaf): Originally, this was a macro defined as follows:
+ *
+ * #define SIGN_EXTEND(a, n) ((((int)a) << (WordSize - (n))) >> (WordSize - (n)))
+ *
```
Comment (kjmcqueen) [inline rosette/src/Ob.cc]: IIRC, shift left shifts in 0's from the right (not relevant here), shift right replicates the top bit. So shifting relevant bits to the left edge of the word then back "sign extends" the field. OTOH, I vaguely remember this being implementation dependent. I'm not sure though.

File: rosette/src/Prim.cc [hunk 2]
```diff
@@ -145,7 +145,14 @@ DEF("runtime-error", obRuntimeError, 1, MaxArgs) {
...
-    return INVALID;
+
+    // NB(leaf): I think it sucks to have a runtime error that can
+    // cause the binary to return 0 to the shell. Aborting here makes
+    // a runtime error into a hard failure, instead of a soft one.
+    // The perror() serves as a reminder to people what the problem
+    // might be.
+    perror("Nota Bene: did you forget to set ulimit -s unlimited?");
+    abort();
 }
```
Comment (kjmcqueen) [inline rosette/src/Prim.cc]: At one point we talked about bringing the ulimit setting internal to Rosette.  Is this possible? Is now the time?

**Comment by kjmcqueen** [comment]
If it still runs boot.rbl and simple tests, I'll happily approve.

File: rosette/src/Prim.cc (hunk 2 above)
Comment (nashef) [reply rosette/src/Prim.cc]: I would rather this wort exist as a reminder to make the GC algorithm not overflow a reasonable stack. This isn't that hard, but today is not the day.

**Comment by nashef** [comment]
Yes, travis runs the tests, including `boot.rbl`.

**Pull request merged**